src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -311,13 +333,11 @@ public CloseableIterator<SearchResult.NodeScore> search(QueryContext context, fl
 
         Bits bits = hasDeletions ? BitsUtil.bitsIgnoringDeleted(toAccept, postingsByOrdinal) : toAccept;
         // VSTODO re-use searcher objects
-        GraphIndex<float[]> graph = builder.getGraph();
-        var searcher = new GraphSearcher.Builder<>(graph.getView()).withConcurrentUpdates().build();
-        NodeSimilarity.ExactScoreFunction scoreFunction = node2 -> {
-            return similarityFunction.compare(queryVector, ((RandomAccessVectorValues<float[]>) vectorValues).vectorValue(node2));
-        };
+        GraphIndex graph = builder.getGraph();
+        var searcher = new GraphSearcher(graph.getView());
+        var ssf = SearchScoreProvider.exact(queryVector, similarityFunction, vectorValues);
         var topK = sourceModel.topKFor(limit, null);
-        var result = searcher.search(scoreFunction, null, topK, threshold, bits);
+        var result = searcher.search(ssf, topK, bits);
```

Comment:
Looks like we lost `threshold` without adding it somewhere else.

Reply:
fixing

Reply:
should i be worried that this isn't exercised by a test?

Reply:
Yes, I'll take a look.

Reply:
Looks like it is because we use graph.size() as the limit: `searchInternal(context, qv, keyRange, graph.size(), threshold)`. I'm not sure that there is a good way test that we pass the threshold.